mm/pmm.c

```diff
@@ -282,7 +282,14 @@ static unsigned find_first_avail_region(void) {
     UNREACHABLE();
 }
 
-static size_t process_memory_range(unsigned index, unsigned first_avail_region) {
+static bool ranges_overlap(paddr_t range_a_start, size_t range_a_size,
+                           paddr_t range_b_start, size_t range_b_size) {
+    return (range_a_start <= (range_b_start + range_b_size)) &&
+           (range_b_start <= (range_a_start + range_a_size));
+}
+
```

Comment:
This function looks like a good member of the `mm/regions.c` family of functions. It could also be based around the `addr_range_t` type.